Yona/Yona/Welcome/WelcomeViewController.swift

```diff
@@ -34,6 +44,59 @@ class WelcomeViewController: BaseViewController {
         setViewControllerToDisplay(ViewControllerTypeString.welcome, key: YonaConstants.nsUserDefaultsKeys.screenToDisplay)
     }
     
+    var didEditTextField = false
+    @objc func textFieldDidChange(textField: UITextField) {
+        self.didEditTextField = true
+    }
+    
+    @IBAction func showEnvironmentSwitchUI(sender: AnyObject) {
```

Comment:
This function is way too long. It needs to be broken down in short methods with self-explanatory names (at most 20 lines long). Remember Uncle Bob:
> Clean code reads like well-written prose

Reply:
Functionality broken down in 2 short methods.